diff --git a/lib/decompress_inflate.c b/lib/decompress_inflate.c
index 555c06b..aa0d8cc 100644
--- a/lib/decompress_inflate.c
+++ b/lib/decompress_inflate.c
@@ -193,7 +193,7 @@ STATIC int INIT gunzip(unsigned char *buf, long len,
 	return __gunzip(buf, len, fill, flush, out_buf, 0, pos, error);
 }
 #else
-STATIC int INIT __decompress(unsigned char *buf, long len,
+int __decompress(unsigned char *buf, long len,
 			   long (*fill)(void*, unsigned long),
 			   long (*flush)(void*, unsigned long),
 			   unsigned char *out_buf, long out_len,
